services/motion/builtin/wrapperFrame.go

```diff
@@ -0,0 +1,148 @@
+	ptgSolvers        []tpspace.PTGSolver
+}
+
+func newWrapperFrame(
+	localizationFrame, executionFrame referenceframe.Frame,
+) (referenceframe.Frame, error) {
+	ptgFrame, ok := executionFrame.(tpspace.PTGProvider)
+	if !ok {
+		return nil, errors.New("cannot type assert executionFrame into a tpspace.PTGProvider")
+	}
+	return &wrapperFrame{
+		name:              executionFrame.Name(),
+		localizationFrame: localizationFrame,
+		executionFrame:    executionFrame,
+		ptgSolvers:        ptgFrame.PTGSolvers(),
+	}, nil
+}
+
+// Name returns the name of the wrapper frame's execution frame's name.
+func (wf *wrapperFrame) Name() string {
+	return wf.name
+}
+
+// Transform returns the associated pose given a list of inputs.
+func (wf *wrapperFrame) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
+	if len(inputs) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(inputs), len(wf.DoF()))
+	}
+	executionFramePose, err := wf.executionFrame.Transform(inputs[:len(wf.executionFrame.DoF())])
+	if err != nil {
+		return nil, err
+	}
+	localizationFramePose, err := wf.localizationFrame.Transform(inputs[len(wf.executionFrame.DoF()):])
+	if err != nil {
+		return nil, err
+	}
+	return spatialmath.Compose(localizationFramePose, executionFramePose), nil
+}
+
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (wf *wrapperFrame) Interpolate(from, to []referenceframe.Input, by float64) ([]referenceframe.Input, error) {
+	if len(from) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(from), len(wf.DoF()))
+	}
+	if len(to) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(to), len(wf.DoF()))
+	}
+	interp := make([]referenceframe.Input, 0, len(to))
+
+	// both from and to are lists with length = 11
+	// the first four values correspond to ptg(execution) frame values we want the interpolate
+	// the latter seven values correspond to pose(localization) frame values we want the interpolate
+
+	// executionFrame interpolation
+	executionFrameFromSubset := make([]referenceframe.Input, len(wf.executionFrame.DoF()))
+	executionFrameToSubset := to[:len(wf.executionFrame.DoF())]
+	interpSub, err := wf.executionFrame.Interpolate(executionFrameFromSubset, executionFrameToSubset, by)
+	if err != nil {
+		return nil, err
+	}
+	interp = append(interp, interpSub...)
+
+	// localizationFrame interpolation
+	// interpolating the localizationFrame is a special case
+	// the ToSubset of the localizationFrame does not matter since the executionFrame interpolations
+	// move us through a given segment and the localizationFrame input values are what we compose with
+	// the position ourselves in our absolute position in world.
+	localizationFrameFromSubset := from[len(wf.executionFrame.DoF()):]
+	interpSub, err = wf.localizationFrame.Interpolate(localizationFrameFromSubset, localizationFrameFromSubset, by)
```

Comment:
Isn't this unnecessary? Since we Know what the localizationFrame always is and how this struct is always used, `interpSub` must always be identical to `localizationFrameFromSubset`.

Reply:
> Isn't this unnecessary?

you're right it is un-needed.
I will update to instead do:
```
interp = append(interp, localizationFrameSubset...)
```